apps/system/js/card.js

```diff
@@ -350,11 +338,111 @@
   };
 
   Card.prototype._fetchElements = function c__fetchElements() {
+    this.contentNode = this.element.querySelector('.content');
     this.screenshotView = this.element.querySelector('.screenshotView');
     this.titleNode = this.element.querySelector('h1.title');
     this.iconButton = this.element.querySelector('.appIcon');
   };
 
+  Card.prototype._registerEvents = function c__registerEvents() {
+    this.element.addEventListener('touchstart', this);
+    this.element.addEventListener('touchmove', this);
+    this.element.addEventListener('touchend', this);
+    this._eventsRegistered = true;
+  };
+
+  Card.prototype._unregisterEvents = function c__unregisterEvents() {
+    if (!this._eventsRegistered || !this.element) {
+      return;
+    }
+    this.element.removeEventListener('touchstart', this);
+    this.element.removeEventListener('touchmove', this);
+    this.element.removeEventListener('touchend', this);
+    this._eventsRegistered = false;
+  };
+
+  Card.prototype.handleEvent = function(evt) {
+    var verticalY;
+    var tapThreshold = 1;
+    switch (evt.type) {
+      case 'touchstart':
+        evt.stopPropagation();
+        this._dragPhase = '';
+        this.deltaX = 0;
+        this.deltaY = 0;
+        this.startTouchPosition = [evt.touches[0].pageX, evt.touches[0].pageY];
+        break;
+
+      case 'touchmove':
+        evt.stopPropagation();
+        this.deltaX = evt.touches[0].pageX - this.startTouchPosition[0];
+        this.deltaY = evt.touches[0].pageY -this.startTouchPosition[1];
+        verticalY = -1 * this.deltaY;
+        console.log('touchmove, y: ', this.deltaY);
+        switch (this._dragPhase) {
+          case '':
+            if (verticalY > Math.abs(this.deltaX) &&
+                verticalY > this.SWIPE_WOBBLE_THRESHOLD) {
+              this._dragPhase = 'cross-slide';
+              // dont try and transition while dragging
+              this.contentNode.style.transitionProperty = 'none';
```

Comment:
I think we want a `transition = 'transform 0s linear'`, from my experience it makes things smoother.